run_cmake.py

```diff
@@ -1,71 +1,57 @@
-# This file is part of Desktop App Toolkit,
-# a set of libraries for developing nice desktop applications.
-#
-# For license and copyright information please follow this link:
-# https://github.com/desktop-app/legal/blob/master/LEGAL
+import sys
+import os
+import shutil
+import subprocess
 
-import sys, os, shutil, subprocess
 
 def run(project, arguments, buildType=''):
     scriptPath = os.path.dirname(os.path.realpath(__file__))
-    basePath = scriptPath + '/../out/' + buildType
+    basePath = os.path.join(scriptPath, '..', 'out', buildType)
 
     cmake = ['cmake']
-    windowsArch = ''
-    explicitGenerator = False
+    cmake_args = []
+
     for arg in arguments:
         if arg == 'debug':
-            cmake.append('-DCMAKE_BUILD_TYPE=Debug')
-        elif arg == 'x86' or arg == 'x64':
-            windowsArch = arg
+            cmake_args.append('-DCMAKE_BUILD_TYPE=Debug')
         elif arg != 'force':
-            if arg.startswith('-G'):
-                explicitGenerator = True
-            cmake.append(arg)
+            cmake_args.append(arg)
+
     if sys.platform == 'win32':
-        if windowsArch == 'x64':
-            cmake.append('-Ax64')
-        else:
-            cmake.append('-AWin32') # default
-    elif windowsArch != '':
-        print("[ERROR] x86/x64 switch is supported only on Windows.")
-        return 1
+        cmake_args.append('-AWin32')
     elif sys.platform == 'darwin':
-        if not explicitGenerator:
-            cmake.append('-GXcode')
-    else:
-        if not explicitGenerator:
-            cmake.append('-GNinja Multi-Config')
-        elif buildType:
-            cmake.append('-DCMAKE_BUILD_TYPE=' + buildType)
-        elif not '-DCMAKE_BUILD_TYPE=Debug' in cmake:
-            cmake.append('-DCMAKE_BUILD_TYPE=Release')
+        cmake_args.append('-GXcode')
+    elif buildType:
+        cmake_args.append('-DCMAKE_BUILD_TYPE=' + buildType)
+    elif '-DCMAKE_BUILD_TYPE=Debug' not in cmake_args:
+        cmake_args.append('-DCMAKE_BUILD_TYPE=Release')
 
     specialTarget = ''
-    specialTargetFile = scriptPath + '/../' + project + '/build/target'
+    specialTargetFile = os.path.join(scriptPath, '..', project, 'build', 'target')
     if os.path.isfile(specialTargetFile):
         with open(specialTargetFile, 'r') as f:
             for line in f:
                 target = line.strip()
                 if len(target) > 0:
-                    cmake.append('-DDESKTOP_APP_SPECIAL_TARGET=' + target)
+                    cmake_args.append('-DDESKTOP_APP_SPECIAL_TARGET=' + target)
 
-    cmake.extend(['-Werror=dev', '-Werror=deprecated', '--warn-uninitialized', '..' if not buildType else '../..'])
-    command = '"' + '" "'.join(cmake) + '"'
+    cmake_args.extend(['-Werror=dev', '-Werror=deprecated', '--warn-uninitialized', '..' if not buildType else '../..'])
+    command = ' '.join(cmake + cmake_args)
```

Comment:
It doesn't really feels like a right thing to do. This `cmake + cmake_args` concatenation has absolutely no sense and what's more important this removes the double quote guards for the arguments.

Reply:
```suggestion
    command = shlex.join(cmake + cmake_args)
```